omod/src/main/java/org/openmrs/module/pihmalawi/web/controller/QuickProgramsFormController.java

```diff
@@ -66,7 +66,7 @@ public ModelAndView enroll(HttpServletRequest request,
 		// enroll patient
 		PatientProgram pp = enrollInProgram(request, patient, program, enrollmentDate, location, null);
 		// set initial state
-		pws.changeToState(pp, state.getProgramWorkflow(), state, enrollmentDate);
+		pp.transitionToState(state, enrollmentDate);
```

Comment:
Have you tested this?  I don't know if this would work, since you are not actually saving the change via programworkflowservice, you are just using a method on the patientprogram to update the property values. You should explicity save the patient program (pws.savePatientProgram(pp) after calling this method.

Reply:
I have updated this code too. Thanks @mseaton !